File: Numerical_Methods/Solving Ordinary Differential Equations (ODEs)/Runge-Kutta Methods (RK)/python_solutions/problem6.py
import numpy as np

def f(x, y):
    return x*x + y*y  # y' = x^2 + y^2

def rk2(x0, y0, h, target_x):
    x = x0
    y = y0
    
    while x < target_x:
        k1 = h * f(x, y)
        k2 = h * f(x + h, y + k1)
        
        y = y + (k1 + k2) / 2
        x = x + h
    
    return y

def main():
    x0 = 0
    y0 = 1
    h = 0.2
    target_x = 0.8
    
    result = rk2(x0, y0, h, target_x)
    print(f"Solution at x = {target_x:.1f}: y = {result:.6f}")
    
    # Note: This problem doesn't have a simple exact solution
    print("Note: This problem doesn't have a simple exact solution")

if __name__ == "__main__":
    main() 

File: Numerical_Methods/Solving Ordinary Differential Equations (ODEs)/Runge-Kutta Methods (RK)/java_solutions/Problem6.java
public class Problem6 {
    public static double f(double x, double y) {
        return x*x + y*y;  // y' = x^2 + y^2
    }

    public static double rk2(double x0, double y0, double h, double targetX) {
        double x = x0;
        double y = y0;
        
        while (x < targetX) {
            double k1 = h * f(x, y);
            double k2 = h * f(x + h, y + k1);
            
            y = y + (k1 + k2) / 2;
            x = x + h;
        }
        
        return y;
    }

    public static void main(String[] args) {
        double x0 = 0;
        double y0 = 1;
        double h = 0.2;
        double targetX = 0.8;
        
        double result = rk2(x0, y0, h, targetX);
        System.out.printf("Solution at x = %.1f: y = %.6f%n", targetX, result);
        
        // Note: This problem doesn't have a simple exact solution
        System.out.println("Note: This problem doesn't have a simple exact solution");
    }
} 